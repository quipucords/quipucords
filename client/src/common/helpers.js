export const bindMethods = (context, methods) => {
  methods.forEach(method => {
    context[method] = context[method].bind(context);
  });
};

export const noop = Function.prototype;
