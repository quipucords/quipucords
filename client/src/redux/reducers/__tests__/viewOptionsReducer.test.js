import viewOptionsReducer from '../viewOptionsReducer';
import {
  viewTypes,
  viewPaginationTypes,
  viewToolbarTypes,
  credentialsTypes,
  scansTypes,
  sourcesTypes
} from '../../constants';
import helpers from '../../../common/helpers';

describe('viewOptionsReducer', () => {
  it('should return the initial state', () => {
    expect(viewOptionsReducer.initialState).toBeDefined();
  });

  it('should handle specific defined types', () => {
    const specificTypes = [
      viewToolbarTypes.SET_FILTER_TYPE,
      viewToolbarTypes.SET_FILTER_VALUE,
      viewToolbarTypes.ADD_FILTER,
      viewToolbarTypes.REMOVE_FILTER,
      viewToolbarTypes.CLEAR_FILTERS,
      viewToolbarTypes.SET_SORT_TYPE,
      viewToolbarTypes.TOGGLE_SORT_ASCENDING,
      viewPaginationTypes.VIEW_FIRST_PAGE,
      viewPaginationTypes.VIEW_LAST_PAGE,
      viewPaginationTypes.VIEW_PREVIOUS_PAGE,
      viewPaginationTypes.VIEW_NEXT_PAGE,
      viewPaginationTypes.VIEW_PAGE_NUMBER,
      viewPaginationTypes.SET_PER_PAGE,
      viewTypes.SELECT_ITEM,
      viewTypes.DESELECT_ITEM,
      viewTypes.EXPAND_ITEM
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        type: value,
        viewType: viewTypes.SOURCES_VIEW
      };

      const resultState = viewOptionsReducer(undefined, dispatched);

      expect({ type: value, result: resultState }).toMatchSnapshot(`defined type ${value}`);
    });
  });

  it('should handle all defined fulfilled types', () => {
    const specificTypes = [credentialsTypes.GET_CREDENTIALS, sourcesTypes.GET_SOURCES, scansTypes.GET_SCANS];

    specificTypes.forEach(value => {
      const dispatched = {
        type: helpers.FULFILLED_ACTION(value),
        payload: {
          data: {
            test: 'success'
          }
        }
      };

      const resultState = viewOptionsReducer(undefined, dispatched);

      expect({ type: helpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot(
        `fulfilled types ${value}`
      );
    });
  });
});
