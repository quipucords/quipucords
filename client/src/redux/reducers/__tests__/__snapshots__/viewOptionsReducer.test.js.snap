// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defined type ADD_FILTER 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [
        undefined,
      ],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "ADD_FILTER",
}
`;

exports[`defined type CLEAR_FILTERS 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "CLEAR_FILTERS",
}
`;

exports[`defined type DESELECT_ITEM 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "DESELECT_ITEM",
}
`;

exports[`defined type EXPAND_ITEM 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "EXPAND_ITEM",
}
`;

exports[`defined type REMOVE_FILTER 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "REMOVE_FILTER",
}
`;

exports[`defined type SELECT_ITEM 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [
        undefined,
      ],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SELECT_ITEM",
}
`;

exports[`defined type SET_FILTER_TYPE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": undefined,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SET_FILTER_TYPE",
}
`;

exports[`defined type SET_FILTER_VALUE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": undefined,
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SET_FILTER_VALUE",
}
`;

exports[`defined type SET_PER_PAGE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": undefined,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SET_PER_PAGE",
}
`;

exports[`defined type SET_SORT_TYPE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": undefined,
      "sortType": undefined,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "SET_SORT_TYPE",
}
`;

exports[`defined type TOGGLE_SORT_ASCENDING 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": false,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "TOGGLE_SORT_ASCENDING",
}
`;

exports[`defined type VIEW_FIRST_PAGE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "VIEW_FIRST_PAGE",
}
`;

exports[`defined type VIEW_LAST_PAGE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 0,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "VIEW_LAST_PAGE",
}
`;

exports[`defined type VIEW_NEXT_PAGE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "VIEW_NEXT_PAGE",
}
`;

exports[`defined type VIEW_PAGE_NUMBER 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "VIEW_PAGE_NUMBER",
}
`;

exports[`defined type VIEW_PREVIOUS_PAGE 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "VIEW_PREVIOUS_PAGE",
}
`;

exports[`fulfilled types GET_CREDENTIALS 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": undefined,
      "totalPages": NaN,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "GET_CREDENTIALS_FULFILLED",
}
`;

exports[`fulfilled types GET_SCANS 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": undefined,
      "totalPages": NaN,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
  },
  "type": "GET_SCANS_FULFILLED",
}
`;

exports[`fulfilled types GET_SOURCES 1`] = `
Object {
  "result": Object {
    "CREDENTIALS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SCANS_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": 0,
      "totalPages": 0,
    },
    "SOURCES_VIEW": Object {
      "activeFilters": Array [],
      "currentPage": 1,
      "expandedItems": Array [],
      "filterType": null,
      "filterValue": "",
      "pageSize": 15,
      "selectedItems": Array [],
      "sortAscending": true,
      "sortField": "name",
      "sortType": null,
      "totalCount": undefined,
      "totalPages": NaN,
    },
  },
  "type": "GET_SOURCES_FULFILLED",
}
`;
